testsuite/run.py

```diff
@@ -139,12 +147,22 @@ def find_testcases():
 )
 
 parser.add_argument(
+    '--verbose', action='store_true',
+    help='Use Valgrind to detect invalid memory operations and leaks'
+)
+
+parser.add_argument(
     'pattern', nargs='*',
     help='List of pattern to filter the set of testcases to run'
 )
 
 
 def main(args):
+
+    if args.verbose:
+        global verbose
```

Comment:
So far, I tried to avoid globals: I just passed “args” to all functions that need to inspect command-line options. Would it be possible to add it as well to the “run_program” function, instead?